### ISSUE TITLE
The variable type rule has an issue when the variable type is a non-null type.

### ISSUE THREAD

**Issue by michaelstaib** [ChilliCream/hotchocolate#262]
**Describe the bug**
The variable type rule has an issue when the variable type is a non-null type.

**To Reproduce**
The following query is rejected by the server:

```GraphQL
query f($x: Episode!){
  hero(episode:$x)
  {
    name
  }
}
```

with the following error:

```json
{
  "errors": [
    {
      "VariableName": "x",
      "Message": "The variable value is not of the correct type."
    }
  ]
}
```

The hero signature looks like the following `hero(episode: Episode!): Character`.

**Expected behavior**
The query should have not been rejected.

**Comment by michaelstaib** [comment]
```csharp

private void CheckForInvalidValueType(in Variable variable)
        {
            if (!variable.Type.IsInstanceOfType(variable.Value))
            {
                throw new QueryException(new VariableError(
                    "The variable value is not of the correct type.",
                    variable.Name));
            }
        }
```

**Comment by michaelstaib** [comment]
This issue is fixed with PR #274
